tests/framework-tests/core/js/FluidJSTests.js

```diff
@@ -391,6 +391,61 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
         jqUnit.assertDeepEq("Array copy", array, copy);
     });
 
+    jqUnit.test("flattenObjectPaths: deep values", function () {
+        var originalObject = {
+            path: {
+                to: {
+                    value: "foo"
+                }
+            },
+            otherPath: {
+                to: {
+                    otherValue: "bar"
+                }
+            }
+        };
+        var output         = fluid.flattenObjectPaths(originalObject);
+        var expectedValue = {
+            "path": "[object Object]",
+            "path.to": "[object Object]",
+            "path.to.value": "foo",
+            "otherPath": "[object Object]",
+            "otherPath.to": "[object Object]",
+            "otherPath.to.otherValue": "bar"
+        };
+        jqUnit.assertDeepEq("Deep values should be flattened correctly.", expectedValue, output);
+    });
+
+    jqUnit.test("flattenObjectPaths: reflattening", function () {
+        var originalObject = { "deep.path.to.value": "foo"};
+        var output = fluid.flattenObjectPaths(originalObject);
+        jqUnit.assertDeepEq("A preflattened object should remain unchanged when flattened again.", originalObject, output);
+    });
+
+    jqUnit.test("flattenObjectPaths: top-level array", function () {
+        var originalObject = ["monkey", "fighting", "snakes"];
+        var output = fluid.flattenObjectPaths(originalObject);
+        jqUnit.assertDeepEq("A top-level array should be flattened correctly.", { 0: "monkey", 1: "fighting", 2: "snakes"}, output);
+    });
+
+    jqUnit.test("flattenObjectPaths: deep array", function () {
+        var originalObject = { "plane": ["monday", "to", "friday"]};
+        var output = fluid.flattenObjectPaths(originalObject);
+        jqUnit.assertDeepEq("A deep array should be flattened correctly.", { "plane": "monday,to,friday", "plane.0": "monday", "plane.1": "to", "plane.2": "friday"}, output);
+    });
+
+    jqUnit.test("flattenObjectPaths: deep empty objects", function () {
```

Comment:
We should add a test to verify that the code doesn't explode with a `null` value as a member. I think, by eye, that it won't, but because it features a typeof === "object" check there is the suggestion of a risk